tests/unit/test_peak_collection.py

```diff
@@ -53,13 +103,45 @@ def test_peak_collection():
     # check effective parameters
     obs_eff_peaks, obs_eff_errors = peaks.get_effective_params()
     assert obs_eff_peaks.size == NUM_SUBRUN
-    np.testing.assert_equal(obs_eff_errors, np.zeros(NUM_SUBRUN, dtype=get_parameter_dtype(effective=True)))
+    np.testing.assert_equal(obs_eff_errors, target_errors)
     np.testing.assert_equal(obs_eff_peaks['Center'], raw_peaks_array['PeakCentre'])
-    np.testing.assert_equal(obs_eff_peaks['Height'], raw_peaks_array['Height'])
-    np.testing.assert_equal(obs_eff_peaks['FWHM'], 2. * np.sqrt(2. * np.log(2.)) * raw_peaks_array['Sigma'])
+    if peak_shape == 'PseudoVoigt':
+        np.testing.assert_equal(obs_eff_peaks['Intensity'], raw_peaks_array['Intensity'])
+        np.testing.assert_equal(obs_eff_peaks['FWHM'], raw_peaks_array['FWHM'])
+        np.testing.assert_equal(obs_eff_peaks['Mixing'], raw_peaks_array['Mixing'])
+    else:
+        np.testing.assert_equal(obs_eff_peaks['Height'], raw_peaks_array['Height'])
+        np.testing.assert_equal(obs_eff_peaks['FWHM'], 2. * np.sqrt(2. * np.log(2.)) * raw_peaks_array['Sigma'])
     np.testing.assert_equal(obs_eff_peaks['A0'], NUM_SUBRUN * [0.])
     np.testing.assert_equal(obs_eff_peaks['A1'], NUM_SUBRUN * [0.])
-    # not checking integrated intensity
+    # check d spacing
+    obs_dspacing, obs_dspacing_errors = peaks.get_dspacing_center()
+    np.testing.assert_allclose(obs_dspacing, target_d_spacing_center, atol=0.01)
+    np.testing.assert_allclose(obs_dspacing_errors, target_d_spacing_center_error, atol=0.01)
+
+
+def test_peak_collection_Gaussian():
+    NUM_SUBRUN = 2
+    # without wavelength
+    check_peak_collection('Gaussian', NUM_SUBRUN, np.zeros(NUM_SUBRUN, dtype=get_parameter_dtype(effective=True)))
+    # with wavelength
+    check_peak_collection('Gaussian', NUM_SUBRUN, np.zeros(NUM_SUBRUN, dtype=get_parameter_dtype(effective=True)),
+                          wavelength=3.15, target_d_spacing_center=[1.57, 1.73],
+                          target_d_spacing_center_error=[0.0, 0.0])
```

Comment:
@peterfpeterson line 126 actually is the test without setting wavelength that would confirm  that the d-spacing returned is nan

Reply:
I completely missed the comment pointing that out :man_facepalming: